ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs

```diff
@@ -55,5 +54,41 @@ public IActionResult Create(CreateVM VM)
             }
             return View(VM);
         }
+
+        public IActionResult Edit(int? id)
+        {
+            var parkingSlot = _slotService.GetById(id);
+            if (parkingSlot is null)
+            {
+                return NotFound();
+            }
+            var VM = new EditVM(parkingSlot);
+            return View(VM);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Edit(int id, EditVM VM)
+        {
+            if(id != VM.Id)
+            {
+                return NotFound();
+            }
+
+            if (ModelState.IsValid) 
+            {
+                var parkingSlot = _slotService.GetById(id);
+                var ParkingSlotExists = _slotService.ParkingSlotExits(VM.ParkingZoneId, VM.Number);
+                if (VM.Number <= 0 || ParkingSlotExists == true && VM.Number != parkingSlot.Number)
+                {
+                    ModelState.AddModelError("Number", "Parking Slot exists or Number is not valid");
+                    return View(VM);
+                }
```

Comment:
lets write that code above if (ModelState.IsValid) in order to avoid nesting (KISS)